app/uk/gov/hmrc/helptosavefrontend/repo/SessionStore.scala

```diff
@@ -0,0 +1,139 @@
+            data ← OptionT.fromOption[EitherStringOr](cache.data)
+            result ← OptionT.liftF[EitherStringOr, HTSSession](
+              (data \ "htsSession").validate[HTSSession].asEither.leftMap(e ⇒ s"Could not parse session data from mongo: ${e.mkString("; ")}"))
+          } yield result
+
+          val _ = timerContext.stop()
+
+          response.value
+
+        }.recover {
+          case e ⇒
+            val _ = timerContext.stop()
+            metrics.sessionReadErrorCounter.inc()
+            Left(e.getMessage)
+        }
+
+      case None ⇒
+        // is this correct to return None if no session found in the HeaderCarrier ? or should this be an Left(error) ?
+        Right(None)
+    })
+  }
+
+  override def store(newSession: HTSSession)(implicit writes: Writes[HTSSession], hc: HeaderCarrier): Result[Unit] = {
+
+      def doUpdate(newSession: HTSSession, oldSession: Option[HTSSession]): Future[Either[String, Unit]] = {
+        val timerContext = metrics.sessionStoreWriteTimer.time()
+        hc.sessionId.map(_.value) match {
+          case Some(sessionId) ⇒
+
+            val sessionToStore = oldSession.fold(
+              newSession
+            )(existing ⇒
+              HTSSession(
+                eligibilityCheckResult = newSession.eligibilityCheckResult.orElse(existing.eligibilityCheckResult),
+                confirmedEmail         = newSession.confirmedEmail.orElse(existing.confirmedEmail),
+                pendingEmail           = newSession.pendingEmail.orElse(existing.pendingEmail),
+                ivURL                  = newSession.ivURL.orElse(existing.ivURL),
+                ivSuccessURL           = newSession.ivSuccessURL.orElse(existing.ivSuccessURL),
+                bankDetails            = newSession.bankDetails,
+                changingDetails        = newSession.changingDetails,
+                accountNumber          = newSession.accountNumber,
+                hasSelectedEmail       = newSession.hasSelectedEmail
+              )
+            )
+
+            doCreateOrUpdate(Id(sessionId), "htsSession", Json.toJson(sessionToStore))
+              .map[Either[String, Unit]] { dbUpdate ⇒
+                if (dbUpdate.writeResult.inError) {
+                  Left(dbUpdate.writeResult.errMsg.getOrElse("unknown error during inserting session data in mongo"))
+                } else {
+                  val _ = timerContext.stop()
+                  Right(())
+                }
+              }.recover {
+                case e ⇒
+                  val _ = timerContext.stop()
+                  metrics.sessionStoreErrorCounter.inc()
+                  Left(e.getMessage)
+              }
+
+          case None ⇒
+            Left("can't store HTSSession in mongo due to no sessionId found in the HeaderCarrier")
+        }
+      }
+
+    for {
+      oldSession ← get
+      result ← EitherT(doUpdate(newSession, oldSession))
+    } yield result
+
+  }
+
+  private[repo] def doFindById(id: Id) =
```

Comment:
this function and `doCreateOrUpdate` are not that useful to have - when reading the code above where they're used, we have to scroll down to see what is happening inside the function only to see that it's only one line of code